### PR TITLE
Add join-str-last and cleanup non-existent functions

### PR DESCRIPTION
**Pull request by sorawee**
This is to reflect changes in https://github.com/brownplt/pyret-lang/pull/1208

### CODE REVIEW

File: src/trove/lists.scrbl [hunk 1]
```diff
@@ -1603,6 +1548,20 @@ end
...
 
 
+@list-method["join-str-last"]
+Combines the values of the current @pyret{List} by converting them to strings
+with @pyret{tostring} and joining them with the given separator @pyret{sep}.
+If the list has more than one element, the function will use @pyret{last-sep}
+to join the last element instead of the regular @pyret{sep}.
+@examples{
+check:
+  [list: 1, 2, 3].join-str-last("; ", "$ ") is "1; 2$ 3"
```
Comment (blerner) [inline src/trove/lists.scrbl]: These examples are deliberately obscure.  Probably a better example would be to produce "1, 2, 3 and 4", or something a bit more human-relevant.

File: src/trove/lists.scrbl [hunk 2]
```diff
@@ -25,30 +25,6 @@
 '(module
   "lists"
   (path "src/arr/base/lists.arr")
-  (unknown-item
```
Comment (blerner) [inline src/trove/lists.scrbl]: These bindings are present in the lists file, at least for now, so technically they should stay here.  The call to `@ignore` below lets us know in this documentation file that we don't intend to explain them.  It's a holdover from when we autogenerated the skeletons of "which functions do I need to document for this file?"

File: src/trove/lists.scrbl [hunk 3]
```diff
@@ -1179,54 +1169,9 @@
...
-        (a-app (a-id "List" (xref "lists" "List")) "a")
-        (a-id "Number" (xref "<global>" "Number"))
-        "Any"
-        "a"))
-    (doc
-      "Returns a new list with the same values as the given list but with the nth element\n        set to the given value, or raises an error if n is out of range"))
   (fun-spec
     (name "length")
     (arity 1))
-  (fun-spec
```
Comment (blerner) [inline src/trove/lists.scrbl]: These bindings, on the other hand -- I'm not sure how they got here.  Did they move from lists.arr to another file?  Anyway it's good you've removed them :)

File: src/trove/lists.scrbl (hunk 1 above)
Comment (blerner) [reply src/trove/lists.scrbl]: You should also have an example of `[list: 1].join-str("sep", "last") is "1"`, similar to your last example below, to drive home the degenerate cases when one or both separators aren't needed.

Comment (sorawee) [reply src/trove/lists.scrbl]: Noted. Will fix.

File: src/trove/lists.scrbl (hunk 2 above)
Comment (sorawee) [reply src/trove/lists.scrbl]: Not anymore :) So I won't do anything regarding this.

File: src/trove/lists.scrbl (hunk 1 above)
Comment (sorawee) [reply src/trove/lists.scrbl]: Fixed

**Pull request merged**